docs/rootless.md

```diff
@@ -77,7 +86,118 @@ PID   USER     TIME   COMMAND
    29 user       0:00 ps aux
 ```
 
+To change the UID, you need to modify and build the BuildKit image manually.
 ```
-$ go get ./examples/build-using-dockerfile
-$ build-using-dockerfile --buildkit-addr tcp://127.0.0.1:1234 -t foo /path/to/somewhere
+$ vi hack/dockerfiles/test.Dockerfile
+$ docker build -t buildkit-rootless-custom --target rootless -f hack/dockerfiles/test.Dockerfile .
 ```
+
+#### Procless mode
+
+Procless mode allows running rootless BuildKit in a container without `--privileged`, by using the BuildKit host PID namespace (i.e. the `buildkitd` container namespace when running in a container).
+Note that Procless mode allows build containers to `kill` (and potentially `ptrace` depending on the seccomp configuration) an arbitrary process in the BuildKit host namespace.
+
+To enable Procless mode, you need to start `buildkitd` with `--oci-worker-procless`.
+You also need to pass `--security-opt seccomp=unconfined --security-opt apparmor=unconfined` to `docker run`.
+
+```
+$ docker run --name buildkitd -d --security-opt seccomp=unconfined --security-opt apparmor=unconfined -p 1234:1234  moby/buildkit:rootless --addr tcp://0.0.0.0:1234 --oci-worker-procless
```

Comment:
Maybe we could provide a seccomp profile that works instead of disabling everything. I'm not sure how much would we need to change(in addition to allowing unprivileged userns clone). This can be follow-up.